### PR TITLE
Updating contributing guide, small fixes, and adding guidance for MapDataPipe

### PR DESCRIPTION
**Pull request by NivekT**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #379

The changes here do not impact any functionalities. Let me know if anything is unclear and I will make additional edits.

Differential Revision: [D36014738](https://our.internmc.facebook.com/intern/diff/D36014738)

### CODE REVIEW

**Comment by NivekT** [comment]
@nivekt has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36014738).

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -82,14 +82,24 @@ conda install -c conda-forge pre-commit
...
+     [`test_serialization.py`](https://github.com/pytorch/data/blob/main/test/test_serialization.py).
+   - If your test requires interacting with files in the file system (e.g. opening a `csv` or `tar` file, we prefer
+     those files to be generated during the test (see `test_local_io.py`). If the file is on a remote server, see
+     `test_remote_io.py`.
 2. Documentation - ensure that the DataPipe has docstring, usage example, and that it is added to the right category of
    the right RST file to be rendered.
+   - If your DataPipe has a functional form (i.e. `@functional_datapipe(...)`), include at the
+     [end of the first sentence](https://github.com/pytorch/data/blob/main/torchdata/datapipes/iter/util/combining.py#L119)
+     of your docstring. This will make sure it correctly shows up in the
+     [summary table](https://pytorch.org/data/beta/torchdata.datapipes.iter.html#archive-datapipes) of our
```
Comment (ejguan) [inline CONTRIBUTING.md:101]: nit: s/beta/main/
beta may change in the future when we reach stable stage.

File: docs/source/torchdata.datapipes.map.rst [hunk 2]
```diff
@@ -4,17 +4,28 @@ Map-style DataPipes
...
 corresponding label from a folder on the disk.
 
 .. autoclass:: MapDataPipe
 
 
+By design, there are fewer ``MapDataPipe`` than ``IterDataPipe`` to avoid duplicate implementations of the same
+functionalities as ``MapDataPipe``. We encourage users to use the built-in ``IterDataPipe`` for various functionalities,
+and convert it to ``MapDataPipe`` as needed using ``MapToIterConverter`` or ``.to_iter_datapipe()``.
+If you have any question or feedback about this design choice, feel free to open or comment on an exising Github issue
+and we will be happy to discuss your use case.
```
Comment (ejguan) [inline docs/source/torchdata.datapipes.map.rst:20]: I thought we wanted to let users to ask questions in the forum

File: torchdata/datapipes/iter/util/unzipper.py [hunk 3]
```diff
@@ -19,7 +19,7 @@ class UnZipperIterDataPipe(IterDataPipe[T]):
     r"""
     Takes in a DataPipe of Sequences, unpacks each Sequence, and return the elements in separate DataPipes
     based on their position in the Sequence. The number of instances produced equals to the sequence legnth
-    minus the number of columns to skip.
+    minus the number of columns to skip (functional name: ``unzip``).
```
Comment (ejguan) [inline torchdata/datapipes/iter/util/unzipper.py:22]: nit: I think you want to move it into the first sentence.

**Review by ejguan — APPROVED** [review]
Thank you. Have a few nit comments

File: docs/source/torchdata.datapipes.map.rst (hunk 2 above)
Comment (NivekT) [reply docs/source/torchdata.datapipes.map.rst:20]: Let me specify that usage question/best practice will be asked in forum, and suggestion to improve our design or adding new `MapDataPipe` should be GH issues

File: torchdata/datapipes/iter/util/unzipper.py (hunk 3 above)
Comment (NivekT) [reply torchdata/datapipes/iter/util/unzipper.py:22]: Good catch!

**Comment by NivekT** [comment]
@nivekt has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36014738).

**Comment by NivekT** [comment]
@nivekt has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36014738).

**Pull request closed without merging**